modules/mig_with_percent/variables.tf

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "hostname" {
+  description = "Hostname prefix for instances"
+  default     = "default"
+}
+
+variable "region" {
+  description = "The GCP region where the managed instance group resides."
+}
+
+variable "instance_template_version0" {
+  description = "Instance template self_link used to create compute instances for initial version"
+}
+
+variable "instance_template_version1" {
+  description = "Instance template self_link used to create compute instances for second version"
+}
+
+variable "version1_percent" {
+  description = "Percentage of instances defined in the second version"
+}
+
+variable "target_size" {
+  description = "The target number of running instances for this managed instance group. This value should always be explicitly set unless this resource is attached to an autoscaler, in which case it should never be set."
+  default     = 1
+}
+
+variable "target_pools" {
+  description = "The target load balancing pools to assign this group to."
+  type        = "list"
+  default     = []
+}
+
+variable "distribution_policy_zones" {
+  description = "The distribution policy, i.e. which zone(s) should instances be create in. Default is all zones in given region."
+  type        = "list"
+  default     = []
+}
+
+#################
+# Rolling Update
+#################
+
+variable "update_policy" {
```

Comment:
Is this actually supported for this module? I thought the balancing mechanism required taking control of the canary updates.

Reply:
I think we should still provide this variable, the update_policy contains some original arguments such as `min_ready_sec `

Reply:
Would we override values in it though?

Reply:
@morgante Sorry, I couldn't understand what you're saying. You meant we should declare some original arguments as variables?

Reply:
Never mind, I see the use case for `min_ready_sec` and similar. Sounds fine.